js-envs/src/test/scala/org/scalajs/jsenv/test/TimeoutComTests.scala

```diff
@@ -57,6 +58,33 @@ trait TimeoutComTests extends TimeoutTests with ComTests {
   }
 
   @Test
+  def receiveTimeoutTest = {
+
+    val com = comRunner(s"""
+      scalajsCom.init(function(msg) {
+        setTimeout(scalajsCom.send, 2000, "Got: " + msg);
+      });
+    """)
+
+    com.start()
+
+    for (i <- 1 to 2) {
+      com.send(s"Hello World: $i")
+      try {
+        com.receive(900.millis)
+        fail("Expected TimeoutException to be thrown")
+      } catch {
+        case _: TimeoutException =>
+      }
+      assertEquals(s"Got: Hello World: $i", com.receive(3000.millis))
```

Comment:
Yep, although in practice it still won't, because sending the fragments will happen in a very short time frame compared to the setTimeouts. So it's basically untestable, unfortunately.